### PR TITLE
Add azure_sql_server(s) resources

### PR DESCRIPTION
**Pull request by omerdemirok**
Signed-off-by: Omer Demirok <odemirok@chef.io>

### Description

BACKEND
- Add `#create_additional_properties` method to make additional API calls simpler
- Improve invalid api version filtering

RESOURCES
- Add fully backward compatible azure_sql_server(s) resources
- Update existing resources with `#create_additional_properties` method

### Issues Resolved

N/A

### Check List

- [x] New functionality includes tests
- [x] All tests pass
- [x] `rake lint` passes
- [x] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco>

### CODE REVIEW

File: libraries/azure_generic_resource.rb [hunk 1]
```diff
@@ -144,4 +144,23 @@ def resource_group
...
   end
+
+  # Create properties on a resource acquired via additional API call in a static method.
+  # @param opts [Hash]
+  #   property_name [string] The name of the property.
+  #   property_endpoint [string] The URI of the properties.
+  #     E.g., '/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/
+  #     Microsoft.Sql/servers/{serverName}/firewallRules'.
+  #   api_version [string] The api version of the endpoint (default - latest).
+  def create_additional_properties(opts = {})
```
Comment (rmoles) [inline libraries/azure_generic_resource.rb:155]: Thoughts on changing the method to `additional_resource_properties` ?

Comment (omerdemirok) [reply libraries/azure_generic_resource.rb:155]: We can update 👍

Comment (omerdemirok) [reply libraries/azure_generic_resource.rb]: Updated.

**Review by rmoles — APPROVED** [review]
Thanks @omerdemirok. Great job.

**Pull request merged**